script/create-build-settings.js

```diff
@@ -13,17 +12,58 @@ function getBuildSettings(options) {
     vic: {
       rateLimitAuthed: 1,
       rateLimitUnauthed: 1
-    }
+    },
+    applications: {}
   };
 }
 
 /**
  * Writes a human-readable JavaScript file containing build properties available globally under `window.settings`.
  * @param {object} options The build options as passed to the build script and processed through Metalsmith.
  */
-module.exports = function writeBuildSettings(options) {
-  const settings = getBuildSettings(options);
-  const settingsPath = path.join(options.destination, 'js/settings.js');
-  const settingsContent = `window.settings = ${JSON.stringify(settings, null, ' ')};`;
-  fs.writeFileSync(settingsPath, settingsContent);
-};
+function createBuildSettings(options) {
+  return (files, metalsmith, done) => {
+    const settings = getBuildSettings(options);
+
+    for (const fileName of Object.keys(files)) {
+      const file = files[fileName];
+      const {
+        entryname: entryName
+      } = file;
+
+      if (!entryName) continue;
+
+      const validTypes = ['string', 'boolean', 'number'];
+      const frontmatter = {};
+
+      for (const dataKey of Object.keys(file)) {
+        const data = file[dataKey];
+        if (validTypes.includes(typeof data)) {
+          frontmatter[dataKey] = data;
+        }
+      }
+
+      let application = settings.applications[entryName];
+      if (!application) {
+        application = {};
+        settings.applications[entryName] = application;
+      }
+
+      application.contentProps = application.contentProps || [];
+      application.contentProps.push(frontmatter);
+    }
+
+    const settingsPath = 'js/settings.js';
+
+    files[settingsPath] = {
+      path: settingsPath,
+      contents: `window.settings = ${JSON.stringify(settings, null, ' ')};`
+    };
+
+    options.settings = settings;
+
+    done();
+  };
+}
+
+module.exports = createBuildSettings;
```

Comment:
This is the logic where the front matter is collected into the data structure. For simplicity, I filtered the data to only allow primitives.